pkg/kclient/kclient.go

```diff
@@ -0,0 +1,53 @@
+package kclient
+
+import (
+	"errors"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// errorMsg is the message for user when invalid configuration error occurs
+const errorMsg = `
+Please ensure you have an active kubernetes context to your cluster. 
+Consult your Kubernetes distribution's documentation for more details
+`
+
+// Client is a collection of fields used for client configuration and interaction
+type Client struct {
+	KubeClient       kubernetes.Interface
+	KubeConfig       clientcmd.ClientConfig
+	KubeClientConfig *rest.Config
+	Namespace        string
+}
+
+// New creates a new client
+func New(skipConnectionCheck bool) (*Client, error) {
+	var client Client
+
+	// initialize client-go clients
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	client.KubeConfig = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+
+	config, err := client.KubeConfig.ClientConfig()
+	if err != nil {
+		return nil, errors.New(err.Error() + errorMsg)
```

Comment:
we have an `errors.Wrap` function in utils which can be used here

Reply:
Thanks, updated!